### PR TITLE
library code migrated to our project

### LINKED ISSUE #2831: Jcenter dependency: com.pacioianu.david:ink-page-indicator
**Describe the bug**
This library is only used for page indicator on intro fragment so i think best is to customised code without using this library since its only used for intro screen. 

This library is also not maintained or available in maven or jitpack.

part of #2816

### PR DESCRIPTION
**Pull request by MohitMaliFtechiz**
Fixes #2831

### CODE REVIEW

File: app/src/main/java/org/kiwix/kiwixmobile/intro/CustomPageIndicator.kt [hunk 1]
```diff
@@ -0,0 +1,808 @@
...
+import java.util.Arrays
+import kotlin.math.abs
+import kotlin.math.max
+import kotlin.math.min
+
+class CustomPageIndicator @JvmOverloads constructor(
+  context: Context,
+  attrs: AttributeSet? = null,
+  defStyle: Int = 0
+) : View(context, attrs, defStyle), OnPageChangeListener, View.OnAttachStateChangeListener {
```
Comment (gouri-panda) [inline app/src/main/java/org/kiwix/kiwixmobile/intro/CustomPageIndicator.kt:49]: We should write documentation about where the code is coming from. It should acknowledge the author(s) and in the future, it'll help us if we encounter bugs :)

Comment (MohitMaliFtechiz) [reply app/src/main/java/org/kiwix/kiwixmobile/intro/CustomPageIndicator.kt:56]: done

**Comment by kelson42** [comment]
@MohitMaliFtechiz Can you please rebase this and @gouri-panda could review it?

**Comment by MohitMaliFtechiz** [comment]
> @MohitMaliFtechiz Can you please rebase this and @gouri-panda could review it?

hi @gouri-panda , rebase done.

File: buildSrc/src/main/kotlin/Libs.kt [hunk 2]
```diff
@@ -15,6 +15,12 @@ object Libs {
   const val select_folder_document_file =
     "androidx.documentfile:documentfile:" + Versions.document_file_version
 
+  /**
+   * https://developer.android.com/jetpack/androidx/releases/swiperefreshlayout#kts
+   */
+  const val swipe_refresh_layout =
+    "androidx.swiperefreshlayout:swiperefreshlayout:" + Versions.swipe_refresh_layout_version
+
```
Comment (gouri-panda) [inline buildSrc/src/main/kotlin/Libs.kt:23]: We have already this dependency.

**Review by gouri-panda — APPROVED** [review]

**Pull request merged**